Tests/Rules/AvoidUsingBrokenHashAlgorithms.tests.ps1

```diff
@@ -0,0 +1,51 @@
+﻿# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+BeforeAll {
+    $settings = @{
+        IncludeRules = @('PSAvoidUsingBrokenHashAlgorithms')
+        Rules        = @{
+            PSAvoidUsingBrokenHashAlgorithms = @{
+                Enable = $true
+            }
+        }
+    }
+
+    $violationMessageMD5 = [regex]::Escape("The Algorithm parameter of cmdlet 'Get-FileHash' was used with the broken algorithm 'MD5'.")
+    $violationMessageSHA1 = [regex]::Escape("The Algorithm parameter of cmdlet 'Get-FileHash' was used with the broken algorithm 'SHA1'.")
+}
+
+Describe "AvoidUsingBrokenHashAlgorithms" {
+    Context "When there are violations" {
+        It "detects broken hash algorithms violations" {
+            (Invoke-ScriptAnalyzer -ScriptDefinition 'Get-FileHash foo -Algorithm MD5' -Settings $settings).Count | Should -Be 1
+            (Invoke-ScriptAnalyzer -ScriptDefinition 'Get-FileHash foo -Algorithm SHA1' -Settings $settings).Count | Should -Be 1
```

Comment:
An optional improvement would be to make it also work when no named parameters are used: `Invoke-ScriptAnalyzer -ScriptDefinition 'Get-FileHash foo SHA1' `

Reply:
Good idea. Can you see of any cases where this might false flag? Ie, someone opening a file that's named the same as a flagged hashing algorithm?

Reply:
No, because what you would do is check whether the first and second item in the array are not named parameters, i.e. do not start with dash and therefore you know for sure that the first one has to be the file name and the second one the algorithm